### ISSUE TITLE
Error when trying to open certain documents with stacked annotations for curation

### ISSUE THREAD

**Issue by s2fische** [inception-project/inception#4835]
It's not possible to open curation mode (via the curation-tab) for one specific document (it does work for other documents). After clicking on the document name, an error code appears (see screenshot) while the rest of the screen stays completely white/blank. Opening the text in curation beta-mode (via the annotation-tab) works, but no annotations are merged in it. The different annotators' annotations are still saved and can be opened in view-only mode.

Our guess is that it might have something to do with having the document still in progress while updating INCEpTION from version 32.1 to 32.2 because only the in-progress document is affected; the ones already marked as finished before the update can be opened in curation mode as usual.

![Clipboard - 29  Mai 2024 15_10](https://github.com/inception-project/inception/assets/170112224/fcc49595-c249-4292-9d28-078d659fce27)

**Please complete the following information:**
 - Version and build ID: 32.2 (2024-05-14 20:46:45, build 101ef653)
 - OS: Windows 10
 - Browser: Chrome

**Comment by reckart** [comment]
This error appears when the system is trying to access an annotation by its ID but the annotation cannot be found. The life cycle of annotation IDs in INCEpTION is very likely not as you might expect from other systems. In fact, these IDs have a rather limited lifetime and are regularly re-generated. So the issue comes up when for some reason a reference to an ID lives longer that the ID itself. Could you please provide a stack trace from the log for this error?

**Comment by s2fische** [comment]
[loginception0529130517.txt](https://github.com/user-attachments/files/15511038/loginception0529130517.txt)

**Comment by s2fische** [comment]
The document itself could suddenly be opened again in curation mode, for some reason (although the error code did still appear).

![Clipboard - 31  Mai 2024 15_33](https://github.com/inception-project/inception/assets/170112224/3ea3baf6-f938-400e-acf4-1f0753d635ce)

However, jumping between different pages (with just one sentence on them) led to problems - at some point, the different annotators' first sentence didn't show up anymore even when it was selected in curation mode (see screenshot).

![Clipboard - 31  Mai 2024 15_38](https://github.com/inception-project/inception/assets/170112224/5008b8bb-9085-4843-b4c6-433ed9bf4c18)

**Comment by reckart** [comment]
Thanks for the report and the logs. I think I have an idea what is going wrong here.

**Comment by reckart** [comment]
@s2fische Maybe you could get the 32.4-SNAPSHOT standalone JAR from here and test it?

https://build.ukp.informatik.tu-darmstadt.de/job/INCEpTION%2032.x/

To make sure the SNAPSHOT version does not interfere with any INCEpTION data you might already have on your machine, best test it on you PC/Laptop (i.e. not on a server) and **start it using a throw-away data directory**, e.g.

```
java -Dinception.home=inception-test-data-directory -jar inception-app-webapp-32.4-SNAPSHOT-standalone.jar
```

Then export the project that is giving your trouble, import it into this test instance, and report back if the problem still occurs.

**Comment by s2fische** [comment]
Thanks, but I can't install it, although I updated java a moment ago

![Clipboard - 4  Juni 2024 14_52](https://github.com/inception-project/inception/assets/170112224/7c61d7f5-93c0-48a0-b3e4-102fad0c3fe6)

**Comment by reckart** [comment]
Probably there is a `JAVA_HOME` environment variable on your system which still points to the Java 8 version that you have installed.

**Comment by bgittel** [comment]
> @s2fische Maybe you could get the 32.4-SNAPSHOT standalone JAR from here and test it?
> 
> https://build.ukp.informatik.tu-darmstadt.de/job/INCEpTION%2032.x/
> 
> To make sure the SNAPSHOT version does not interfere with any INCEpTION data you might already have on your machine, best test it on you PC/Laptop (i.e. not on a server) and **start it using a throw-away data directory**, e.g.
> 
> ```
> java -Dinception.home=inception-test-data-directory -jar inception-app-webapp-32.4-SNAPSHOT-standalone.jar
> ```
> 
> Then export the project that is giving your trouble, import it into this test instance, and report back if the problem still occurs.
> 
> After the test, you can remove the `inception-test-data-directory` folder that was created in the working directory.

The problem seems to be gone in the 32.4-SNAPSHOT standalone (I'm in the same team with s2fische). 

To avoid problems of this kind in the future, I'd like to know, if it is not advisable to update Inception on the server, while there are started, but unfinished documents? We were suspecting that this caused the error, because only the document where annotators had started before the update, showed the strange behavior in the curation.

**Comment by reckart** [comment]
> To avoid problems of this kind in the future, I'd like to know, if it is not advisable to update Inception on the server, while there are started, but unfinished documents? We were suspecting that this caused the error, because only the document where annotators had started before the update, showed the strange behavior in the curation.

The issue had nothing to do with the state of the document. 

The bug was introduced by the attempt to fix #4741. Unfortunately, that fix was buggy itself.

You should not have to worry about the state of documents when upgrading to a new INCEpTION version.

**Comment by bgittel** [comment]
Great, thank you!
